api/handlers.go

```diff
@@ -239,22 +239,27 @@ func (a *API) Mount(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Only perform the TTL refresh if the driver is in unlocked mode.
-	if !volConfig.Unlocked {
-		if err := a.startTTLRefresh(volName); err != nil {
-			a.HTTPError(w, err)
+	mc, err := driver.Mount(driverOpts)
+	if err != nil {
+		if err := a.Lock.ClearLock(ut, (*a.Global).Timeout); err != nil {
+			a.HTTPError(w, errors.RefreshMount.Combine(errored.New(volConfig.String())).Combine(err))
 			return
 		}
-	}
 
-	mc, err := driver.Mount(driverOpts)
-	if err != nil {
 		a.HTTPError(w, errors.MountFailed.Combine(err))
 		return
 	}
 
 	a.MountCollection.Add(mc)
 
+	// Only perform the TTL refresh if the driver is in unlocked mode.
+	if !volConfig.Unlocked {
+		if err := a.startTTLRefresh(volName); err != nil {
+			a.HTTPError(w, err)
```

Comment:
@erikh Do we ever get into this situation here? If it does, the published lock dangles again right ?